client/src/components/Navbar.js

```diff
@@ -40,44 +40,61 @@ function UserAuthNav(props) {
   );
 }
 
-const NavBar = props => {
-  // used "!!" to cast variable to boolean. Used to avoid the behavior of undefined in
-  // conditional logic. In brief !undefined = true > !true = false
-  const userIsLoggedIn = props.auth && (!!props.auth.googleID || !!props.auth.facebookID);
-  let homeText;
-  props.location.pathname === '/' ? (homeText = 'RE:IMAGINE DENVER') : (homeText = 'HOME');
-  return (
-    <Navbar bsStyle="remove-default" collapseOnSelect fluid>
-      <Navbar.Header>
-        <Navbar.Brand>
-          <Link to="/">{homeText}</Link>
-        </Navbar.Brand>
-        <Navbar.Toggle />
-      </Navbar.Header>
-      <Navbar.Collapse>
-        <Nav pullRight>
-          <NavItem eventKey={1}>
-            <Link to="/denver-learn-more">Why</Link>
-          </NavItem>
-          <NavDropdown id="tools-dropdown" eventKey={2} title="Tools">
-            <MenuItem eventKey={2.1}>
-              <Link to="/manager-resources">Property Manager Resources</Link>
-            </MenuItem>
-            <MenuItem eventKey={2.2}>
-              <Link to="/tips-for-requesting">Tips for Requesting</Link>
-            </MenuItem>
-          </NavDropdown>
-          <NavItem eventKey={3}>
-            <Link to="/who-are-we">Who Are We</Link>
-          </NavItem>
-          {/*  RENDERS MyCampaignNavItem BASED ON AUTH STATUS */}
-          {userIsLoggedIn && <MyCampaignNavItem />}
-          {/* UserAuthNav (Login/Logout) */}
-          <UserAuthNav auth={props.auth} logOutUser={props.logSignerOut} />
-        </Nav>
-      </Navbar.Collapse>
-    </Navbar>
-  );
+class NavBar extends React.Component {
+  componentWillReceiveProps = nextProps => {
```

Comment:
Consider using shouldComponentUpdate() here. This will be deprecated soon and we might be able to prevent unnecessary re-renders.

[https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops](url)

Reply:
shouldComponentUpdate() returns a boolean that decides if a re-render is necessary.  componentWillReceiveProps is the time to set up state.  Since this logic does not concern re-rendering but does concern state I think it is best to keep using this method for now.

When we do upgrade to react 16 we will be able to use the new getDerivedStateFromProps() to fill its place.